core/src/main/java/org/hibernate/ogm/dialect/impl/DotPatternMapHelpers.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.dialect.impl;
```

Comment:
It'd be better located in `org.hibernate.ogm.datastore.document.impl`.

Btw. to give some background, the "*.spi" packages are meant to provide the required types/functionality for implementing dialects. We give some guarantees regarding compatible evolvement there. Other things where we don't feel like committing to that yet, we have under "impl" and still refer them from "our" dialects. The idea is to promote those things to "spi" if they have proven over time. This helper also feels like it's rightly located under "impl" for now and once we see it fits for different use cases / dialects, we can relocate it in a subsequent release.